tdms/tests/unit/array_tests/test_XYZVectors.cpp

```diff
@@ -0,0 +1,53 @@
+/**
+ * @file test_XYZVectors.cpp
+ * @author William Graham
+ * @brief Tests allocation, deallocation, and methods for the XYZVectors class
+ */
+#include <catch2/catch_test_macros.hpp>
+#include <spdlog/spdlog.h>
+
+#include "arrays.h"
+#include "globals.h"
+
+const double tol = 1e-8;
+
+TEST_CASE("Test XYZVectors allocation") {
+  SPDLOG_INFO("== Testing XYZVectors allocation/deallocation");
+  // dimensions for the test-tensor
+  const int nx = 4, ny = 8, nz = 16;
+
+  // try creating a XYZVector
+  XYZVectors v;
+  // check that, although this has been declared, its members still point to nullptrs
+  REQUIRE(v.x == nullptr);
+  REQUIRE(v.y == nullptr);
+  REQUIRE(v.z == nullptr);
+
+  // create some arrays to assign to the members
+  double x_vec[nx] = {0.}, y_vec[ny] = {2., 1.}, z_vec[nz] = {4., 3., 2., 1.};
+  // and for testing they're pointing to the right values
+  double x_tot = 0., y_tot = 0., z_tot = 0.;
+
+  // now let's try assigning
+  v.set_ptr(AxialDirection::X, x_vec);
+  REQUIRE(v.x != nullptr);
+  v.set_ptr(AxialDirection::Y, y_vec);
+  REQUIRE(v.y != nullptr);
+  v.set_ptr(AxialDirection::Z, z_vec);
+  REQUIRE(v.z != nullptr);
+  // and manipulating the components
+  for (int ix = 0; ix < nx; ix++) { x_tot += v.x[ix]; }
```

Comment:
```suggestion
  double x_tot = 0.;
  for (int ix = 0; ix < nx; ix++) { x_tot += v.x[ix]; }
```
For clarity, I might have declared the total(s) just here above the loop. Then delete space so l27 and lines 31-37 are all part of the same "block".

Reply:
I wondered also, about using `std::accumulate` here. Might simplify a bit.